src/main/scala/org/scalameter/Measurement.scala

```diff
@@ -1,15 +1,19 @@
 package org.scalameter
 
+import org.scalameter.picklers.Pickler
 import scala.collection.Seq
 
 @SerialVersionUID(-2541697615491239986L)
-case class Measurement(value: Double, params: Parameters, data: MeasurementData, units: String) {
-  def complete: Seq[Double] = data.complete
+case class Measurement[T: Pickler](value: T, params: Parameters,
+  data: MeasurementData[T], units: String) {
+  private[scalameter] def pickler: Pickler[T] = implicitly[Pickler[T]]
+
+  def complete: Seq[T] = data.complete
   def success: Boolean = data.success
-  def errors: Errors = new Errors(this)
-  def failed = this.copy(data = data.copy(success = false))
+  def errors(implicit num: Numeric[T]): Errors[T] = new Errors(this)
```

Comment:
It looks like `errors` is not used anywhere.
Having `errors` in fact pollutes everything with the `Numeric` typeclass.

I would just remove it altogether. Those values in `errors` would only be used by warmers, measurers and reporters that know that the result type is numeric, so they would have the numeric typeclass anyway.